### ISSUE TITLE
Poor performance and abnormal arm behavior (backward rotation) with fine-tuned GR00T-N1.6 on PnPAppleToPlate

### ISSUE THREAD

**Issue by Xueyangwang-will** [NVIDIA/Isaac-GR00T#574]
I am experiencing significant issues when evaluating the official fine-tuned GR00T-N1.6 model on the PnPAppleToPlate task. The performance is consistently poor, and specifically, the left arm repeatedly rotates backwards in an unnatural manner during the rollout.

Reproduction Script
I am using the following command for testing:
gr00t/eval/sim/GR00T-WholeBodyControl/GR00T-WholeBodyControl_uv/.venv/bin/python gr00t/eval/rollout_policy.py \
    --n_episodes 5 \
    --max_episode_steps=1440 \
    --env_name gr00tlocomanip_g1_sim/LMPnPAppleToPlateDC_G1_gear_wbc \
    --n_action_steps 20 \
    --n_envs 5 \
    --model_path nvidia/GR00T-N1.6-G1-PnPAppleToPlate
Observed Behavior
Success Rate:0
Artifacts: The left arm consistently exhibits backward rotation throughout the episodes.

https://github.com/user-attachments/assets/473052d4-ac90-48e8-83fb-a4ab15dd9265

**Comment by bailu-obj** [comment]
same here, with larger n_action_steps(30) and max_episode_steps(2000), I can get success rate at 0.2 at most, but the result still unusable for real deploy. my model server is running on a jetson thor DK with modified env(update pytorch to 2.10-cu130 version which can install on thor with pip), my WholeBodyControl is running on another x86 rtx3080 PC with official  env.

**Comment by pedrosett** [comment]
We ran into the exact same issue. 0/11 success rate on PnPAppleToPlate using the official pipeline (100% NVIDIA code, main branch at 7d5a455).

Same symptoms: locomotion and balance work fine, left arm goes backward (saturates at ~154°), right arm moves in the right direction but too slowly, hands never activate.

We did an exhaustive investigation ruling out: Isaac-GR00T version, camera rendering, action denormalization ranges, joint ordering (43 DOF, URDF sorted), and WBC submodule commit. Everything checks out, the pipeline is correct.

Our conclusion is that the published checkpoint was likely trained on an internal version of the pipeline that differs from the public one. 

We're currently fine-tuning GR00T N1.6 from the base model (GR00T-N1.6-3B) using NVIDIA's published 102-episode sim dataset and launch_finetune.py, on a B200 GPU (still waiting in queue for access). The idea is to ensure train and eval use the exact same public code. Will report back with results once we have them.

**Comment by pedrosett** [comment]
With NVIDIA's published checkpoint, we also observed a 0% success rate on this task, along with the same abnormal arm behavior shown in this issue.
After training our own version, the G1 reached a 50% success rate on pick-and-place, up from 0% with the published checkpoint.The Hugging Face checkpoint may not correspond to the actual fine-tuned weights.

Videos from our runs:

https://github.com/user-attachments/assets/f9ea40a7-9fcd-411e-b4b6-be1485d5c118

https://github.com/user-attachments/assets/0e37fdde-20d8-4f69-801d-8ad62e4b8a6d

https://github.com/user-attachments/assets/53795b16-9357-4352-a7ad-7150cf3dd9ff

If there is interest, I can share the fine-tuned model used in the videos above.

**Comment by dk-teknologisk-bnbm** [comment]
@pedrosett It would be amazing if you could share the fine-tuned models you used. I'm currently having the issue that the thread describes and would like to test before running my own fine tuning.

**Comment by bailu-obj** [comment]
Hi guys, we have tried to checkout the git commit to `4e62473d5226c55784697944a5c9606a51927bfc` (which is GR00T N1.6 Release commit), then the default model and simulator works fine. Hope this can help you.

**Comment by pedrosett** [comment]
great info @bailu-obj tnx, gonna tgry it here. Just one moment then, i have to upload the files to hugginface. Will upload all the saved chackpoints. brb

**Comment by pedrosett** [comment]
https://huggingface.co/pedroset/GR00T-N1.6-G1-PnPAppleToPlate-CW

**Comment by rylinjames** [comment]
the rollback to 4e62473 works but the underlying issue is the published checkpoint was trained on an internal pipeline that diverged from the public code. @pedrosett's finding confirms it -- retraining from base GR00T-N1.6-3B on the public 102-episode dataset gets you to 50% which is what the checkpoint should have been.

one thing worth checking before retraining: action denormalization. if the normalization stats baked into the checkpoint don't match what the public eval script expects, you'll get the exact symptom here -- arms moving in wrong directions or wrong magnitudes even when the policy itself is fine. we ran into similar issues building reflex-vla's export pipeline (github.com/FastCrest/reflex-vla) where we had to carefully verify the pre/postprocessor json files were being loaded correctly at inference time.

if you're still seeing issues after the rollback, worth dumping the raw action outputs before denormalization and checking if the magnitudes look sane.
